TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/Index.cshtml.cs

```diff
@@ -1,10 +1,158 @@
+using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.Xrm.Sdk.Query;
+using TeachingRecordSystem.Core.Dqt.Models;
+using TeachingRecordSystem.Core.Dqt.Queries;
 
 namespace TeachingRecordSystem.SupportUi.Pages.Persons;
 
-public class IndexModel : PageModel
+public partial class IndexModel : PageModel
 {
-    public void OnGet()
+    private const int MaxSearchResultCount = 500;
+    private const int PageSize = 25;
+
+    [GeneratedRegex("^\\d{7}$")]
+    private static partial Regex TrnRegex();
+
+    private readonly TrsLinkGenerator _linkGenerator;
+    private readonly ICrmQueryDispatcher _crmQueryDispatcher;
+
+    public IndexModel(
+        TrsLinkGenerator linkGenerator,
+        ICrmQueryDispatcher crmQueryDispatcher)
+    {
+        _linkGenerator = linkGenerator;
+        _crmQueryDispatcher = crmQueryDispatcher;
+    }
+
+    [BindProperty(SupportsGet = true)]
+    [Display(Name = "Search")]
+    public string? Search { get; set; }
+
+    [FromQuery(Name = "PageNumber")]
+    public int? PageNumber { get; set; }
+
+    public PersonInfo[]? SearchResults { get; set; }
+
+    public int TotalKnownPages { get; set; }
+
+    public int? PreviousPage { get; set; }
+
+    public int? NextPage { get; set; }
+
+    public async Task<IActionResult> OnGet()
+    {
+        PageNumber ??= 1;
+
+        if (!string.IsNullOrEmpty(Search))
+        {
+            if (PageNumber < 1)
+            {
+                return BadRequest();
+            }
+
+            return await PerformSearch();
+        }
+
+        return Page();
+    }
+
+    public async Task<IActionResult> OnPost()
+    {
+        PageNumber ??= 1;
+
+        if (!ModelState.IsValid)
+        {
+            return this.PageWithErrors();
+        }
```

Comment:
Do we need this check?

Reply:
probably not as there a no potential input errors

Reply:
Removed